Units/parser-php.r/run-guest-nested.d/args.ctags

```diff
@@ -0,0 +1,4 @@
+--sort=yes
+--extras=+rg
+--fields=+rln
+--language-force=html
```

Comment:
This is not a requirement for merging. Just an idea.

I wonder if we can remove --language-force=html.
If the PHP parser recognizes the input is an HTML file when reading "<HTML", the PHP parser can run the HTML parser as a guest. Then the HTML parser runs PHP parser when reading '<?php'

Reply:
Unfortunately, when PHP is the root parser, it does not find the JavaScript functions.  I don't see any way to fix that in the PHP parser, because it only understands "chunks between PHP code", and cannot treat the entire <script> block as a single unit.

Reply:
I inspected the old code that I almost forgot and your change.
In the current implementation, the PHP parser makes a promise for each HTML fragment.
A javascript block is splitted into HTML fragments. As the result, especially the bottom half of the block cannot be parsed well by JavaScript parser.

Instead of making small promises, we can make the PHP parser makes one large promise when "<HTML" is found WITH masking <?php...> and <%...%> areas by whitespace.

A promise can be built with a filter that overwrites the input stream before passing it to a guest parser as far as the filter doesn't change the length of the input stream.
`attachPromiseModifier` defined in promise.c is the engine applying such a filter.
Let me try to apply this algorithm to delete --language-force=html option:-P.

Reply:
Fypp parser is line-oriented but implements this technique; making a promise for whole the file area to run Fortran parser as a guest.
Line filters attached to the promise mask the preprocessor lines in the area with whitespaces.